compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -1204,6 +1204,7 @@ function inferBlock(
         };
         break;
       }
+      case "MetaProperty":
```

Comment:
hmmm this isn't safe: the `new.target` meta-property may refer to a mutable object (constructor function), so we can't consider this value to be immutable. If we want to do this, we'll need to restrict support to just `import` meta properties which shouldn't be modified

Reply:
Great feedback, I'll look into this.

Reply:
I tried addressing this in 3e3b0174c2 and I'm not sure if I correctly handled the case when the property is no from `import.meta`. Can you look into that?

Reply:
Awesome! On my phone but this looks right, will do a proper review next week